src/types/route.ts

```diff
@@ -8,9 +11,9 @@ export type RouteItemDef = {
    * Screen (or component) to show
    * when navigating to the menu item
    */
-  component: React.ReactNode;
-  /** Layout will be used for this route */
-  layout?: React.ReactNode;
+  component: ComponentType<RouteComponentProps>;
+  /** Layout used for this route */
+  layout?: FC;
```

Comment:
should we enforce an `FC` for layout? Making it a little less specific will enable a little more flexibility. Though what exactly is the difference between `component` and `layout` here?

Reply:
`component` is the screen or component to be shown for the path, and the `layout` is a component that we can wrap around our screen, so in the RouteWrapper we have set the default layout to be `<DefaultLayout>`, so we don't have to manually wrap every single screen in the DefaultLayout. But for our Auth screens we usually want a different layout, so that's why we have it. 
Maybe we can solve it in a different way? but I have had projects where the layout changes for different screens, so this approach makes it easy to set a different layout if necessary.